next/src/form.ts

```diff
@@ -94,19 +95,37 @@ function validationErrorsToFormErrors(errors: ValidationErrorWithMessage[]): For
 
     // Clean the path to remove intermediate composition structures
     const cleanedPath = cleanErrorPath(path)
-
-    // For all paths, recursively build the nested structure
     let current = result
 
     // Process all segments except the last one (which will hold the message)
     cleanedPath.slice(0, -1).forEach((segment) => {
-      // If this segment doesn't exist yet or is currently a string (from a previous error),
-      // initialize it as an object
-      if (!(segment in current) || typeof current[segment] === 'string') {
-        current[segment] = {}
-      }
+      // Special case for group-arrays where the next segment is `items` followed by the index of the item
+      if (cleanedPath[1] === 'items' && typeof cleanedPath[2] === 'number') {
+        const [arrayName, _, arrayIndex] = cleanedPath
+
+        if (!(arrayName in result) || typeof result[arrayName] === 'string') {
+          result[arrayName] = [] as Array<null | FormErrors>
+        }
+
+        const array = result[arrayName] as Array<null | FormErrors>
+
+        // Create a new object for this index if it doesn't exist
+        if (array[arrayIndex] === null || array[arrayIndex] === undefined) {
```

Comment:
nitpick: if you prefer a squeezed version of this line, you can use `array[arrayIndex] == null`